### PR TITLE
add missing dependencies to DataFormats/TrackReco

### PR DESCRIPTION
**Pull request by davidlange6**
identified by modules build debugging. I've just done a grep #include on interface/* and src/*

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: DataFormats/TrackReco/BuildFile.xml [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 <use name="DataFormats/MuonDetId"/>
-<use name="DataFormats/BeamSpot"/>
+<use name="DataFormats/SiPixelCluster"/>
 <use name="DataFormats/SiPixelDetId"/>
+<use name="DataFormats/SiStripCluster"/>
 <use name="DataFormats/SiStripDetId"/>
-<use name="DataFormats/SiPixelCluster"/>
-<use name="DataFormats/TrackingRecHit"/>
+<use name="DataFormats/TrackCandidate"/>
+<use name="DataFormats/TrackReco"/>
```
Comment (Dr15Jones) [inline DataFormats/TrackReco/BuildFile.xml:14]:
```suggestion
```

Comment (davidlange6) [reply DataFormats/TrackReco/BuildFile.xml:14]: whoops!

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical: it just adds missing dependencies
- Jenkins tests pass

**Pull request merged**